DemoPage/examples/PrimaryNav/index.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+import React  from 'react'
+import PrimaryNav from '../../../layout/PrimaryNav'
+import PrimaryNavLink from '../../../layout/PrimaryNav/PrimaryNavLink'
+import PrimaryNavLinkSeperator from '../../../layout/PrimaryNav/PrimaryNavLinkSeperator'
+import Icon from '../../../atoms/Icon'
+
+export default React.createClass({
+  displayName: 'PrimaryNavExample',
+
+  getInitialState: function() {
+    return {
+      reduce: false,
+      specialFirstItem: true
+    };
+  },
+
+  toggleReduce: function() {
+    this.setState({
+      reduce: !this.state.reduce
+    });
+  },
+
+  toggleSpecialFirstItem: function() {
+    this.setState({
+      specialFirstItem: !this.state.specialFirstItem
+    });
+  },
+
+  render: function() {
```

Comment:
Not sure if it is in the linter yet but in JS guild we've discussed favouring the ES6 shorthand for methods on objects. So, instead of this:

```js
render: function() {
  // code n' stuff
}
```

You should write:

```js
render() {
  // code n' stuff
}
```

This can be applied to all of your methods.